gui/system/forms.py

```diff
@@ -812,30 +805,25 @@ def done(self, form_list, **kwargs):
             if not share:
                 continue
 
+            kwargs = {}
             share_name = share.get('share_name')
             share_purpose = share.get('share_purpose')
-            share_user = share.get('share_user')
-            share_group = share.get('share_group')
-            share_mode = share.get('share_mode')
+            kwargs['user'] = share.get('share_user')
+            kwargs['group'] = share.get('share_group')
+            kwargs['mode'] = share.get('share_mode')
 
             if share_purpose == 'iscsitarget':
                 continue
 
-            if share_purpose == 'afp':
-                share_acl = 'mac'
-            elif share_purpose == 'cifs':
-                share_acl = 'windows'
+            if share_purpose == 'cifs':
+                kwargs['mode'] = None
             else:
-                share_acl = 'unix'
-
-            _n.mp_change_permission(
-                path='/mnt/%s/%s' % (volume_name, share_name),
-                user=share_user,
-                group=share_group,
-                mode=share_mode,
-                recursive=False,
-                acl=share_acl,
-            )
+                kwargs['acl'] = []
+
+            with client as c:
+                dataset = c.call('pool.dataset.query', [['mountpoint', '=', f'/mnt/{volume_name}/{share_name}']], {'get': True})
```

Comment:
I just noticed this being for the Wizard, there is nothing that assures the path is a dataset, so this might raise an IndexError?

Reply:
Good point, I'll fix that.